tools/code-tools/jcstress.sh

```diff
@@ -36,29 +54,25 @@ detectJdks
 pushd $REPO_DIR
   git checkout master
   tip=`git log | head -n 1 | sed "s/.*\s\+//"` || true
-  tip_shortened=`echo ${tip:0:10}`
+  tip_shortened=`echo ${tip:0:7}`
   latestRelease=`git tag -l | sort -Vr | head -n 1`
   rc=$main_name-$latestRelease.jar
 
   # latest released
-  git checkout $latestRelease
-  export JAVA_HOME=$jdk11
-  mvn clean install
-  mv tests-all/target/$main_name.jar $rc
+  buildJcstress "${jdk11}" "${latestRelease}" "${rc}"
   echo "Manually renaming $rc as $main_file to provide latest-stable-recommended file"
   ln -fv $rc $main_file
-  mvn clean
 
   # tip
-  git checkout master
-  export JAVA_HOME=$jdk11
-  mvn clean install
-  mv tests-all/target/$main_file $main_name-$tip_shortened.jar
+  buildJcstress "${jdk17}" "master" "${main_name}-${tip_shortened}.jar"
   echo "Manually renaming $main_name-$tip_shortened.jar as $main_name-tip.jar to provide latest-unstable-recommended file"
   ln -fv $main_name-$tip_shortened.jar $main_name-tip.jar
-  mvn clean
 
-  echo "Resetting repo back to master"
-  git checkout master
+  # 20240222
```

Comment:
What is the significance of that date?  Is that a tag for a major version?

Reply:
As I see it, there is no significance. Unluckily there is also no tag. Doule unluckily jcstress tags extremely rarely:(

As described here: https://github.com/adoptium/aqa-tests/issues/5261 ;  the jcstres in aqavit have heavily unclear binaries to execute:

> The getDepenencies is pulling jcstress-tests-all-20240222.jar'
https://github.com/adoptium/TKG/blob/master/scripts/getDependencies.pl#L182
in addition, it is pulling it from personal server

and

> however, the https://github.com/adoptium/aqa-tests/blob/master/system/jcstress/playlist.xml
is using jcstress-tests-all-20220908.jar

Those are not base don any tag, nor hash.  They will be reused in the cstress runner, so the  https://github.com/adoptium/TKG/blob/master/scripts/getDependencies.pl#L182 and  https://github.com/adoptium/aqa-tests/blob/master/system/jcstress/playlist.xml are in harmony.

I think first step will be to let the getDependencies.pl to download 20220908, and use that in playlist.xml. Then to bump them both to 20240222 (as this is quite fresh). There may come som if jdk 8/11/...22 swithc as is done with jtrregs. This is stil to be decided. For now, I wanted to prepare binareis to use.

Thanx for bringing it up. If you have any top level hints, dont hesitate to write them to https://github.com/adoptium/ci-jenkins-pipelines/issues/1016 and  https://github.com/adoptium/aqa-tests/issues/5261.
Actually.. any hints welcomed :) Thanx for eyball!

Reply:
@smlambert - I think I need your guidance here :-).  Which version(s) of jcstress does the testing group need?  I read the other issues but I'm still unsure...

Reply:
as any builder, we need tip and latest release for sure.   I heavily recommend to not get stick to latest release, as they are not fluently released, and to verify jcstress failure you have to try something newer, always, even tip from time to time.   So A middle path is to pick up something known to be stable on the fly, and starting using that. 20240202 is pretty fresh, and known to be stable, so very good candidate.